ingestion/src/kvkafka.rs

```diff
@@ -244,21 +247,33 @@ async fn read_kafka(group_name: String, tx: mpsc::Sender<Msg>, cancel_token: Can
             poll_result = async { consumer.poll() } => {
                 match poll_result {
                     Ok(sets) => {
+                        // used for metrics
+                        let mut num_messages = 0;
+
                         for msgset in sets.iter() {
                             for msg in msgset.messages() {
+                                num_messages += 1;
                                 if let Err(e) = parse_message(msg.value, &tx).await {
-                                    eprintln!("{}", e);
+                                    metrics::counter!("kafka_failures").increment(1);
+                                    error!("{}", e);
                                 }
                             }
                             if let Err(e) = consumer.consume_messageset(msgset) {
-                                eprintln!("{}", e);
+                                metrics::counter!("kafka_failures").increment(1);
+                                error!("{}", e);
```

Comment:
Should "{}" be replaced with e.g. "consumer.consume_messageset(msgset): {}" for easier identification of where the error occurred? (ditto for other "anonymous" occurrences of error!).

Reply:
Done